app/app.html

```diff
@@ -0,0 +1,17 @@
+<ion-menu id="leftMenu" [content]="content">
+
+  <ion-toolbar>
+    <ion-title>Pages</ion-title>
+  </ion-toolbar>
+
+  <ion-content>
+    <ion-list>
+      <button ion-item *ngFor="#p of pages" (click)="openPage(p)">
```

Comment:
There’s a surprising amount of Angular 2 syntax crammed into this little line. This will create a button for each page we defined in the constructor earlier (like ng-repeat in Angular 1), the * means it will create an embedded template, and by using #p we are able to keep a reference to a specific page, which we pass into the openPage function (defined in our root component) when the button is clicked. If you go back and take a look at the openPage function again, you will see that it takes this reference and uses it to set the rootPage https://github.com/telecom-se/ionic2-starter-tutorial-annotated/pull/1/files#diff-2de9eda1655d499e934e7465e87ca0bdR54